### PR TITLE
Find HTML comment above <script>

### PR DESCRIPTION
**Pull request by garlicnation**

### CODE REVIEW

File: hydrolysis-analyzer.html [hunk 1]
```diff
@@ -91,6 +91,7 @@
       var options = {
         clean:  this.clean,
         filter: this.transitive ? function() { return false; } : null,
+        attachAst: true,
```
Comment (justinfagnani) [inline hydrolysis-analyzer.html]: what does attachAst mean?

File: lib/ast-utils/docs.js [hunk 2]
```diff
@@ -107,7 +107,7 @@ function annotateElementHeader(descriptor) {
  */
 function annotateElement(descriptor) {
   if (!descriptor.desc && descriptor.type === 'element') {
-    descriptor.desc = _findElementDocs(descriptor.is, descriptor.domModule);
+    descriptor.desc = _findElementDocs(descriptor.is, descriptor.domModule, descriptor.scriptElement);
```
Comment (justinfagnani) [inline lib/ast-utils/docs.js]: line length

File: lib/ast-utils/docs.js [hunk 3]
```diff
@@ -319,41 +319,51 @@ function cleanProperty(property) {
...
-  }
+function _findElementDocs(elementId, domModule, scriptElement) {
   // Note that we concatenate docs from all sources if we find them.
   var found = [];
 
-  // Do we have a HTML comment on the `<dom-module>`?
+  // Do we have a HTML comment on the `<dom-module>` or `<script>`?
   //
   // Confusingly, with our current style, the comment will be attached to
   // `<head>`, rather than being a sibling to the `<dom-module>`
```
Comment (justinfagnani) [inline lib/ast-utils/docs.js]: what's this? does this apply to `<script>` as well?

File: hydrolysis-analyzer.html (hunk 1 above)
Comment (garlicnation) [inline hydrolysis-analyzer.html]: It hints hydrolysis to add AST references to a variety of places. I am convinced that there aren't any use cases that would want this off, now that I've tossed naive JSON serializability out of the window.

File: lib/ast-utils/docs.js (hunk 3 above)
Comment (garlicnation) [inline lib/ast-utils/docs.js]: Due to the rules around where HTML comments end up in the AST, we have to go on a wild goose chase to find element docs. They can be: 
1) A sibling to the dom-module and part of body
2) The last child of head
2) the last child of the document

File: lib/ast-utils/docs.js (hunk 2 above)
Comment (garlicnation) [inline lib/ast-utils/docs.js]: Fixed.

File: lib/ast-utils/docs.js (hunk 3 above)
Comment (justinfagnani) [inline lib/ast-utils/docs.js]: are there parse5 rules, or html rules? can we file bugs?

**Comment by justinfagnani** [comment]
lgtm

File: lib/ast-utils/docs.js (hunk 3 above)
Comment (garlicnation) [inline lib/ast-utils/docs.js]: HTML rules.

**Pull request merged**